smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/DocumentShapeSerdeGenerator.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.typescript.codegen.integration;
+
+import software.amazon.smithy.model.shapes.CollectionShape;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.shapes.UnionShape;
+import software.amazon.smithy.typescript.codegen.integration.ProtocolGenerator.GenerationContext;
+
+/**
+ * Smithy protocol document body shape generators.
+ */
+public interface DocumentShapeSerdeGenerator extends HttpValueProvider {
+
+    /**
+     * Writes the code needed to serialize a structure in the document of a request.
+     *
+     * <p>Implementations of this method are expected to generate a function body that
+     * returns a value representing the StructureShape {@code shape} parameter that is
+     * serializable by {@code serializeDocument}.
+     *
+     * <p>The function signature for this body will have two parameters:
```

Comment:
Should this further clarify "will have two parameters:" -> "will have two parameters available in scope:"